functions/src/dashboard.ts

```diff
@@ -0,0 +1,236 @@
+import * as functions from 'firebase-functions'
+import * as admin from 'firebase-admin'
+
+import * as jwt from 'jsonwebtoken'
+import UserModel from './models/User'
+import {
+  Action,
+  ApplicationStep,
+  User,
+  UserResponse,
+  Path,
+  Step,
+} from './models/Dashboard'
+import { StepStatus } from './models/StepStatus'
+
+import { withEnableCORS } from './utils/decorators'
+import { StepType } from './models/StepType'
+
+const EXCEEDED_DEADLINE_DESC =
+  'Sorry, you have exceeded your deadline. ' +
+  'So you cannot go further into this application 😢'
+
+interface UserStepAndAction {
+  userSteps: Step[]
+  userAction?: Action
+}
+
+function validateJwtAndGetUser(
+  token: string,
+  secretKey: string
+): UserModel | undefined {
+  try {
+    const decodedUser = jwt.verify(token, secretKey, {
+      ignoreExpiration: false,
+      maxAge: '1d',
+    })
+    return decodedUser as UserModel
+  } catch (e) {}
```

Comment:
Differentiate when secret key is not valid (decode) and token expired. In the case when the secret key is not valid, we should Log this.